File: swift/read_file.swift
import Foundation

let filePath = "./my-text-file.txt"
let fileURL = URL(fileURLWithPath: filePath)

let content = try String(contentsOf: fileURL, encoding: .utf8)
let lines = content.components(separatedBy: .newlines)
for line in lines {
    if !line.isEmpty {
        print("line: \(line)")
    }
}

File: rust/read_file.rs
use std::fs::File;
use std::io::{self, BufRead, BufReader};

fn main() {
    let file = File::open("./my-text-file.txt").unwrap();

    for line in BufReader::new(file).lines() {
        println!("line: {}", line.unwrap());
    }
}
